src/plugins/console/public/application/containers/editor/monaco/monaco_editor.tsx

```diff
@@ -6,13 +6,119 @@
  * Side Public License, v 1.
  */
 
-import React, { FunctionComponent, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { CodeEditor } from '@kbn/code-editor';
 import { css } from '@emotion/react';
 import { CONSOLE_LANG_ID } from '@kbn/monaco';
+import { parse } from 'query-string';
+import { debounce } from 'lodash';
+import { decompressFromEncodedURIComponent } from 'lz-string';
+import { i18n } from '@kbn/i18n';
+import { useServicesContext } from '../../../contexts';
+
+export interface EditorProps {
+  initialTextValue: string;
+}
+
+interface QueryParams {
+  load_from: string;
+}
+
+const DEFAULT_INPUT_VALUE = `# Click the Variables button, above, to create your own variables.
+GET \${exampleVariable1} // _search
+{
+  "query": {
+    "\${exampleVariable2}": {} // match_all
+  }
+}`;
+
+export const MonacoEditor = ({ initialTextValue }: EditorProps) => {
+  const {
+    services: {
+      history,
+      notifications,
+      settings: settingsService,
+      http,
+      autocompleteInfo,
+      storage,
+    },
+  } = useServicesContext();
+  const [value, setValue] = useState(initialTextValue);
+
+  useEffect(() => {
+    const readQueryParams = () => {
+      const [, queryString] = (window.location.hash || window.location.search || '').split('?');
+
+      return parse(queryString || '', { sort: false }) as Required<QueryParams>;
+    };
+
+    const loadBufferFromRemote = (url: string) => {
+      // Normalize and encode the URL to avoid issues with spaces and other special characters.
+      const encodedUrl = new URL(url).toString();
+      if (/^https?:\/\//.test(encodedUrl)) {
+        const loadFrom: Record<string, any> = {
```

Comment:
I'm not sure the variable `loadFrom` is used later in this function, I think we could remove it

Reply:
Ah I missed one code block in that if-statement, nice catch! Here we parse the `load_from` param if it is an http - I added this to the test instructions in the PR.